crates/transaction-pool/src/validate/task.rs

```diff
@@ -91,7 +92,10 @@ impl TransactionValidationTaskExecutor<()> {
     }
 }
 
-impl<Client, Tx> TransactionValidationTaskExecutor<EthTransactionValidator<Client, Tx>> {
+impl<Client, Tx> TransactionValidationTaskExecutor<EthTransactionValidator<Client, Tx>>
+where
+    Client: BlockReaderIdExt,
```

Comment:
Ideally we'd condition this trait limitation to the cfg optimism, but it might be a chore

Reply:
Since traits are intended to be additive supporting this would most likely look like introducing an optimism-specific trait type. I think this alternative is easier and doesn't bound `Client` to too much additional functionality. I would suggest we leave as is and rewind it if/when needed, hiding it behind the feature flag.